packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -549,8 +580,34 @@ class SyncWatcher extends BaseWatcher {
         deadline,
         transferSentTxHash: transactionHash,
         transferSentBlockNumber: blockNumber,
-        transferSentIndex
-      })
+        transferSentIndex,
+        isFinalized
+      }
+
+      // When a transfer is finalized, reset its error states. It might have been marked
+      // as notFound previously if there was weird behavior onchain after this
+      // transfer was seen at the head. If this is the case, the transfer would not have been
+      // bonded before finality and will need to be bonded now.
+
+      // NOTE: There is a rare race condition where an unfinalized transfer may be processed before
+      // the finalized transfer is seen but sent and fail onchain validation after the finalized
+      // transfer has been recorded. For example, if a reorg happens at block 255 and finalization
+      // is at block 256. In this case, the transferId will be marked as notFound and the
+      // transfer will never be bonded. This is not handled, but if it ever occurs in practice, handle here.
+      if (isFinalized) {
+        dbData.isNotFound = undefined
+        dbData.withdrawalBondTxError = undefined
+        dbData.withdrawalBondBackoffIndex = undefined
+        logger.debug('finalized transfer seen, resetting unfinalized error states')
+      }
+
+      await this.db.transfers.update(transferId, dbData)
+
+      // Experimental: compare data against WS cache and clear the memory
+      if (this.wsCache[transferId]) {
+        this.compareWsCache(event)
+        this.wsCache[event.args.transferId] = undefined
```

Comment:
suggestion: use `delete` to remove reference from the object for potentially faster garbage collection

```suggestion
delete this.wsCache[event.args.transferId] 
```